flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphGenerator.java

```diff
@@ -221,6 +238,42 @@ public StreamGraph generate() {
 		return builtStreamGraph;
 	}
 
+	private ScheduleMode determineScheduleMode(GlobalDataExchangeMode exchangeMode) {
+		if (scheduleMode != null) {
```

Comment:
A `checkNotNull` is needed here for the argument `exchangeMode` to make sure that the `exchangeMode` is already set before setting the `shuffleMode`.